### PR TITLE
DPC-4433 Tag Repository

### PR DESCRIPTION
**Pull request by jdettmannnava**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4433

## 🛠 Changes

- Added utility workflow tag_release

## ℹ️ Context

We are migrating from Jenkins to github actions for deployment. The first application to implement this functionality is the static site. Part of deployment involves tagging releases, which all releases require. I put the utility for tagging releases in dpc-app because it feels more central than dpc-static-site.

This workflow duplicates the tagging part of [Jenkinsfile.release_static_site](https://github.com/CMSgov/dpc-ops/blob/main/jenkins_files/Jenkinsfile.release_static_site)

## 🧪 Validation

Tagging ran successfully: https://github.com/CMSgov/dpc-static-site/actions/runs/12377126243/job/34545914298

### CODE REVIEW

File: .github/workflows/tag_release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        default: 'main'
+        type: 'string'
+    outputs:
+      tag:
+        description: the release tag
+        value: ${{ jobs.tag_repo.outputs.next_rev }}
+jobs:
+  tag_repo:
+    name: Tag Repository
+    runs-on: ubuntu-latest
```
Comment (chris-ronning-ny) [inline .github/workflows/tag_release.yml:18]: You're sure we don't want to run it on `self-hosted`?

Comment (jdettmannnava) [reply .github/workflows/tag_release.yml:18]: @gsf , @christopher-maboh : Is there a reason to run self-hosted vs ubuntu? I assumed to run on ubuntu-latest unless we needed secrets

Comment (gsf) [reply .github/workflows/tag_release.yml]: As noted in Slack, default to self-hosted for now.

Comment (jdettmannnava) [reply .github/workflows/tag_release.yml]: Switched to self-hosted per: https://cmsgov.slack.com/archives/C04UG13JF9B/p1735574668101029?thread_ts=1735574363.374849&cid=C04UG13JF9B

**Review by MEspositoE14s — APPROVED** [review]
LGTM 👍

**Pull request merged**